elytron/ELY-816-masked-password.adoc

```diff
@@ -0,0 +1,137 @@
+= ELY-816 Support for masked passwords in client XML config
+:author:            Jan Kalina
+:email:             jkalina@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          comma,separated,tags
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse
+
+== Overview
+
+The Elytron client should support masked passwords - passwords encrypted using a predefined key, as it is supported in
+PicketBox. It should be possible to use masked passwords in place of any clear password written in the Elytron Client XML
+configuration.
+
+For example, instead of:
+[source,xml]
+----
+<credentials>
+    <clear-password password="password"/>
+</credentials>
+----
+
+The following would also be possible to use:
+
+[source,xml]
+----
+<credentials>
+    <masked-password iteration-count="12" salt="12345678" masked-password="j37uUs8kG9t2QSWjoxxtDg=="/>
+</credentials>
+----
+
+It should also be possible to specify a masked password to be used for authentication programmatically:
+
+[source,Java,options=nowrap]
+----
+ AuthenticationConfiguration maskedConfig =
+                AuthenticationConfiguration.empty()
+                        .setSaslMechanismSelector(SaslMechanismSelector.NONE.addMechanism(PLAIN))
+                        .useName(USERNAME)
+                        .usePassword("YFBlotObdCo=", null, null, "100", "12345678", null); //attributes of a masked password
+
+----
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-816[ELY-816]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1065[EAP7-1065]
+* https://issues.jboss.org/browse/ELY-867[ELY-867]
+* https://issues.jboss.org/browse/WFLY-12241[WFLY-12241]
+
+
+=== Dev Contacts
+
+*  mailto:aabdelsa@redhat.com[Ashley Abdel-Sayed]
+
+=== QE Contacts
+
+* mailto:istraka@redhat.com[Ivan Straka]
+
+=== Testing By
+[x] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* WildFly Elytron
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
```

Comment:
What about `oauth2-bearer-token` ?

see https://docs.jboss.org/author/display/WFLY/WildFly+Client+Configuration#WildFlyClientConfiguration-%3Ccredentialstores%2F%3E

Reply:
Added as part of the requirements